lib/matplotlib/backends/backend_agg.py

```diff
@@ -367,23 +371,21 @@ def post_processing(image, dpi):
         The saved renderer is restored and the returned image from
         post_processing is plotted (using draw_image) on it.
         """
-
-        width, height = int(self.width), int(self.height)
-
-        buffer, (l, b, w, h) = self.tostring_rgba_minimized()
+        buf = np.asarray(self.buffer_rgba())
+        sly, slx = cbook._get_nonzero_slices(buf[..., 3])
+        buf_min = buf[sly, slx]
```

Comment:
```suggestion
        buffer = np.asarray(self.buffer_rgba())
        slice_y, slice_x = cbook._get_nonzero_slices(buffer[..., 3])
        clipped_buffer = buffer[slice_y, slice_x]
```
For my taste `buf`, `sly`, `slx`, `buf_min` are unclear names because they are too much abbreviated.

(Also needs renaming of the variables in the following code-.)

Reply:
Looks good except for the variable names. Maybe add a test for `_get_nonzero_slices()`.

Reply:
actually it's an image at that point so I renamed to `img`/`cropped_img`.
_get_nonzero_slices is indirectly tested by all its users so I'm not going to bother testing it separately.